### ISSUE TITLE
Certificate and CSR do not match when trying to provide signed certificate

### ISSUE THREAD

**Issue by gregory-schiano** [canonical/manual-tls-certificates-operator#77]
#### Describe the bug
I got a certificate signed by IS based on the CSR using Digicert, I have the following failure: "Certificate and CSR do not match." when running the provide action

#### To Reproduce
Run:
`juju run-action stg-tls-certificate/leader provide-certificate   relation-id=32   certificate="$(base64 -w0 certificate.pem)"   ca-chain="$(base64 -w0 ca-chain.pem)"   ca-certificate="$(base64 -w0 ca-certificate.pem)"   certificate-signing-request="$(base64 -w0 csr.pem)"`
With a certificate signed by DigiCert from a CSR generated by the charm

#### Expected behavior
Certificate is accepted

#### Environment
- Certificate signed by DigiCert provided by IS

#### Additional context
I'm pretty sure the issue comes from here:
https://github.com/canonical/manual-tls-certificates-operator/blob/main/lib/charms/tls_certificates_interface/v2/tls_certificates.py#L1688-L1689

Basically the subject on the CSR and certificates differs because the CSR doesn't any Country, Location and Organisation, and the signing authorities always add them to the signed certificates
The CSR has:
        Subject: CN = iam.identity.staging.canonical.com, x500UniqueIdentifier = 61bb6883-458b-4515-9a80-2da6c786eb37
And certificate has:
        Subject: C = GB, L = London, O = CANONICAL GROUP LIMITED, CN = iam.identity.staging.canonical.com
That can't work :confused:
I checked that the CSR and the certificates matches by checking the modulus which is I think a good way

**Comment by gruyaume** [comment]
Thanks for reporting this, I've added the issue to our next pulse.
